Source/Tools/FEXServer/CoreFileWriter/CoreFileWriter.cpp

```diff
@@ -0,0 +1,865 @@
+        auto &Note = Notes.emplace_back(NoteEntry{
+          .Name = Name,
+          .Type = Type,
+        });
+
+        Note.Data.resize(sizeof(T));
+        T* NoteDataBacking = reinterpret_cast<T*>(Note.Data.data());
+        *NoteDataBacking = NoteData;
+        return &Note;
+      }
+
+      NoteEntry *Note_prstatus{};
+      NoteEntry *Note_prpsinfo{};
+      NoteEntry *Note_siginfo{};
+      NoteEntry *Note_file{};
+      NoteEntry *Note_fpregset{};
+      NoteEntry *Note_xstate{};
+      NoteEntry *Note_auxv{};
+  };
+
+  std::pair<std::optional<fextl::string>, int> CoreFileWriter::OpenCoreDumpFile() {
+    if ((CoreDumpFilter & CoreDumpFilterBits::DUMP_STYLE_MASK) == CoreDumpFilterBits::DUMP_DISABLE) {
+      // Return early if dumping is disabled.
+      return {std::nullopt, -1};
+    }
+
+    fextl::string CoreDumpFolder{};
+    if (CoreDumpConfigFolder().empty()) {
+      CoreDumpFolder = fextl::fmt::format("{}/coredump", FEXCore::Config::GetDataDirectory());
+    }
+    else {
+      CoreDumpFolder = CoreDumpConfigFolder();
+    }
+
+    if (FHU::Filesystem::CreateDirectories(CoreDumpFolder)) {
+      const auto CoreDumpFile = fextl::fmt::format("{}/core.{}.{}.{}.{:%a %Y-%m-%d %H:%M:%S %Z}.coredump",
+        CoreDumpFolder,
+        ApplicationName,
+        UID,
+        PID,
+        fmt::localtime(Timestamp)
+      );
+
+      int FD = open(CoreDumpFile.c_str(), O_CREAT | O_WRONLY | O_TRUNC, 0644);
+      return {CoreDumpFile, FD};
+    }
+
+    return {std::nullopt, -1};
+  }
+
+  bool CoreFileWriter::CompressCoredump(const std::string_view File) {
+    if (!CompressZSTD()) {
+      // If not enabled then don't even try compressing.
+      return false;
+    }
+
+    const auto CompressedCoredump = fextl::fmt::format("{}.zstd", File);
+    const auto CompressCommand = fextl::fmt::format("zstd -q -q --no-progress \"{}\" -o \"{}\"", File, CompressedCoredump);
+
+    // Don't ignore SIGCHLD here. We inherited this from our parent.
+    signal(SIGCHLD, SIG_DFL);
+
+    int pid = fork();
+    if (pid == 0) {
+      // Child
+      const char *argv[8];
+      argv[0] = "zstd";
+      argv[1] = "-q";
+      argv[2] = "-q";
+      argv[3] = "--no-progress";
+      argv[4] = File.data();
+      argv[5] = "-o";
+      argv[6] = CompressedCoredump.c_str();
+      argv[7] = nullptr;
+
+      // Try and execute {erofsfuse, squashfuse} to mount our rootfs
+      if (execvpe(argv[0], (char * const*)argv, environ) == -1) {
+        // End the child
+        exit(1);
+      }
+    }
+    else {
+      // Parent
+      // This will happen with execvpe of squashmount or exit on failure
+      int WaitStatus{};
+      int Ret = waitpid(pid, &WaitStatus, 0);
+
+      if (Ret != -1 && WIFEXITED(WaitStatus) && WEXITSTATUS(WaitStatus) == 0) {
+        LogMan::Msg::IFmt("Wrote Coredump to {}", CompressedCoredump);
+        return true;
+      }
+
+    }
+    return false;
+  }
+
+  void CoreFileWriter::CleanupOldCoredumps() {
```

Comment:
Do we really need this? `/tmp` seems to be a more suitable location for these, and it gets automatically cleaned up by the system so we wouldn't need to bring our own system for that.